src/common/personinfo.js

```diff
@@ -167,8 +167,34 @@ let personinfo = function (tour, aConfig) {
   });
 };
 
+
+const PERSISTENT_LOCALSTORAGE_KEY="localStorageUsed";
+let tryLocalStorage = function (store=localStorage) {
+  try {
+    var haveChecked = JSON.parse(store.getItem(PERSISTENT_LOCALSTORAGE_KEY) || 0);
+    if (!haveChecked || haveChecked < 2) {
+      store.setItem(PERSISTENT_LOCALSTORAGE_KEY, ++haveChecked);
+      return;
+    }
+  } catch (e) {
+    store.setItem(PERSISTENT_LOCALSTORAGE_KEY, 1);
+  }
+};
+
+let isLocalStoragePersistent = function (store=localStorage) {
+  var haveChecked = store.getItem(PERSISTENT_LOCALSTORAGE_KEY) || "0";
+  try {
+    return JSON.parse(haveChecked) >= 2;
```

Comment:
Nit: both of these functions should do the same things inside/outside the try...catch. Here you're fetching the item outside, in tryLocalStorage you're fetching it inside the try block. I don't mind which way around, but the behaviour should be consistent. Note that localStorage.get/setItem can throw in reality, if the db is missing or corrupt. Might be worth handling that in a followup?